vms/platformvm/state/disk_staker_diff_iterator.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+)
+
+const (
```

Comment:
nit: It would be nice to get a comment providing a visual understanding of how these are mapped out over state

Reply:
I unified the BLS key and weight key logic. Also added a couple comments explaining what the keys look like